src/vos/tests/vts_aggregate.c

```diff
@@ -354,7 +354,8 @@ aggregate_basic(struct io_test_args *arg, struct agg_tst_dataset *ds,
 	assert_non_null(buf_u);
 
 	for (epoch = epr_u->epr_lo; epoch <= epr_u->epr_hi; epoch++) {
-		if (punch_nr > 0 && punch_epoch[punch_idx] == epoch) {
+		if (punch_nr > 0 && punch_idx < punch_nr
+		    && punch_epoch[punch_idx] == epoch) {
```

Comment:
This is conflicting with the fix in Jeff's patch. Since 'punch_idx' is always >= 0, so checking 'punch_idx < punch_nr' is enough, checking 'punch_nr > 0' is redundant.